### DISCUSSION TITLE
Intelephense Formatting Badly

### DISCUSSION THREAD

**Discussion by josephcrawfordSRH** [LazyVim/LazyVim#3176] [Q&A]
I am getting so frustrated with the formatting using intelephense and I am not sure if this is the right place to ask this question but I followed the setup from the lazyvim docs.

When I format on save or with <leader>cf it formats my if blocks in such and ugly manner.

This code below will look like this before safe.
```php
if (!empty($request->variableNameHere) && $obj->field === Class::CONSTANT_HERE &&
    Carbon::now()->timezone($timezone)->format('m/d/Y') !== Carbon::parse($request->requestDate)->format('m/d/Y')
) {
```

After the save it does this.
```php
if (
    !empty($request->variableNameHere) && $obj->field === Class::CONSTANT_HERE &&
    Carbon::now()->timezone($timezone)->format('m/d/Y') !== Carbon::parse($request->requestDate)->format('m/d/Y')
) {
```

This happens for MANY if statements but not all.  I do not like the `if (` being on it's own line.  I have checked my ft plugin and set `vim.bo.textwidth = 120` thinking that it was wrapping limits but that did nothing.  I need to get this solved as it is trying to reformat every file i touch.  I do not see a configuration file to tell intelephense how to format my code either.

This is my lspconfig settings for intelephense

```lua 
servers = {
  intelephense = {
    filetypes = { "php", "blade" },
    settings = {
      intelephense = {
        format = {
          braces = "k&r",
        },
        filetypes = { "php", "blade" },
        files = {
          associations = { "*.php", "*.blade.php" }, -- Associating .blade.php files as well
          maxSize = 5000000,
        },
      },
    },
  },
}
```
If anyone has any ideas I would greatly appreciate some tips/suggestions on how I could resolve this.

**Comment by mikedfunk** [comment]
You probably won't like this, but a better solution would be to use phpcbf or php-cs-fixer or laravel pint to use a configuration-based approach to fixing code.

Also:

```php
$shouldDoThing = !empty($request->variableNameHere) && $obj->field === Class::CONSTANT_HERE &&
    Carbon::now()->timezone($timezone)->format('m/d/Y') !== Carbon::parse($request->requestDate)->format('m/d/Y');
    
if ($shouldDoThing) {
    // ...
}
```

This is much easier to read for future you and other developers. (where `shouldDoThing` is replaced with something more specific of course).

**Reply by josephcrawfordSRH** [reply]
thanks I have used those in the past, will look into getting them setup.

**Comment by ronald2wing** [comment]
Try this https://github.com/LazyVim/LazyVim/pull/6830
